src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -1053,15 +1069,24 @@ internal virtual void SaveLastDebuggerAgentBufferReceivedToContext(SessionId ses
         {
         }
 
-        internal async Task<bool> OnReceiveDebuggerAgentEvent(SessionId sessionId, JObject args, CancellationToken token)
+        internal async Task<Result> GetLastDebuggerAgentBuffer(SessionId sessionId, JObject args, CancellationToken token)
         {
-            Result res = await SendMonoCommand(sessionId, MonoCommands.GetDebuggerAgentBufferReceived(RuntimeId), token);
-            SaveLastDebuggerAgentBufferReceivedToContext(sessionId, res);
-            if (!res.IsOk)
+            var argsNew = JObject.FromObject(new
+            {
+                objectId = args?["callFrames"]?[0]?["scopeChain"]?[0]?["object"]?["objectId"]?.Value<string>(),
+            });
+            Result res = await SendCommand(sessionId, "Runtime.getProperties", argsNew, token);
+            return res;
+        }
+
+        internal async Task<bool> OnReceiveDebuggerAgentEvent(SessionId sessionId, JObject args, Result debuggerAgentBuffer, CancellationToken token)
+        {
+            SaveLastDebuggerAgentBufferReceivedToContext(sessionId, debuggerAgentBuffer);
```

Comment:
I see that this has an implementation only for firefox, but why does it need to save it?

Reply:
Because from firefox we receives 2 different messages, one with the pause and another asking for the callstack.